internal/services/apimanagement/api_management_diagnostic_resource.go

```diff
@@ -113,7 +113,6 @@ func resourceApiManagementDiagnostic() *pluginsdk.Resource {
 			"operation_name_format": {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
-				Default:  string(apimanagement.OperationNameFormatName),
```

Comment:
it would be a breaking change to remove the default here as this could break things for existing users who are using this default. However, we could potentially use the features.FourPointOhBeta() flag here so that the remove default can be optionally used. There are some examples of the flag being used in the contributors guide [here](features.FourPointOhBeta())